### PR TITLE
feat(transactions): app_start_type reading/writing

### PR DESCRIPTION
**Pull request by philipphofmann**
Adds reading and writing to the app_start_type . For more context, see https://github.com/getsentry/sentry/issues/36927. This PR is based on the migration PR https://github.com/getsentry/snuba/pull/3124.

PR in Sentry to expose this via Discover. https://github.com/getsentry/sentry/pull/38548.

### CODE REVIEW

File: tests/test_snql_api.py [hunk 1]
```diff
@@ -575,6 +575,30 @@ def test_suspect_spans_data(self) -> None:
         assert len(data) == 1
         assert data[0]["array_spans_exclusive_time"] > 0
 
+    def test_app_start_type(self) -> None:
```
Comment (philipphofmann) [inline tests/test_snql_api.py:578]: @evanh, I'm unsure if this is the best way to test this new column. Can you maybe guide me in the right direction, please?

Comment (evanh) [reply tests/test_snql_api.py:578]: This looks good to me!

File: snuba/datasets/entities/discover.py [hunk 2]
```diff
@@ -284,6 +284,7 @@ def attempt_map(
                 ]
             ),
         ),
+        ("app_start_type", String(Modifiers(nullable=True))),
```
Comment (evanh) [inline snuba/datasets/entities/discover.py:287]: Can you add this to the transactions entity as well please? https://github.com/getsentry/snuba/blob/master/snuba/datasets/entities/transactions.py

File: snuba/datasets/transactions_processor.py [hunk 3]
```diff
@@ -248,6 +248,12 @@ def _process_contexts_and_user(
             if context in contexts:
                 del contexts[context]
 
+        appContext = contexts.get("app")
+        if appContext is not None:
+            appStartType = appContext.get("start_type")
+            if appStartType is not None:
+                processed["app_start_type"] = appStartType
```
Comment (evanh) [inline snuba/datasets/transactions_processor.py:255]: Snuba will implicitly populate a `None` value for any missing columns, but I would rather be as explicit as possible for clarity's sake. Can you set `processed["app_start_type"]` to `None` by default?

File: snuba/datasets/entities/discover.py (hunk 2 above)
Comment (philipphofmann) [reply snuba/datasets/entities/discover.py:287]: Yes, I added it, but I'm unsure if I did it correctly. Why is `transaction_source` not in the transaction entity?

**Review by evanh — APPROVED** [review]

File: snuba/datasets/entities/discover.py (hunk 2 above)
Comment (philipphofmann) [reply snuba/datasets/entities/discover.py:287]: I added a parameterized test.

**Review by nikhars — APPROVED** [review]
LGTM. Please merge during SnS working hours 9AM-5PM PST

File: snuba/datasets/transactions_processor.py (hunk 3 above)
Comment (philipphofmann) [reply snuba/datasets/transactions_processor.py:256]: @evanh, I can confirm that setting `processed["app_start_type"] = None` fails the tests. I don't know how to fix this, as I have no clue how Snuba works, and I also don't have Python knowledge. Can't I remove it here? Do I need to change things somewhere else?

**Comment by lynnagara** [comment]
@philipphofmann Is this supposed to be merged?

**Comment by philipphofmann** [comment]
> LGTM. Please merge during SnS working hours 9AM-5PM PST

Yes, please, you can merge it @lynnagara. Due to the comment above, I didn't merge it during my working hours, and I didn't manage to get it done during 9AM-5PM PST. So anyone in PST, please merge this when you see it.

**Pull request merged**